lib/ancestry/instance_methods.rb

```diff
@@ -288,16 +284,10 @@ def ancestry_callbacks_disabled?
 
   private
 
-    def cast_primary_key(key)
-      if [:string, :uuid, :text].include? primary_key_type
-        key
-      else
-        key.to_i
-      end
-    end
-
-    def primary_key_type
-      @primary_key_type ||= column_for_attribute(self.class.primary_key).type
+    def parse_ancestry_column obj
```

Comment:
Nice!  You were able to remove a method and is still pretty readable.  Both good things!  👍